pkg/kritis/cron/cron.go

```diff
@@ -19,32 +19,96 @@ package cron
 import (
 	"context"
 	"time"
+
+	"github.com/grafeas/kritis/pkg/kritis/apis/kritis/v1beta1"
+	"github.com/grafeas/kritis/pkg/kritis/crd/securitypolicy"
+	"github.com/grafeas/kritis/pkg/kritis/metadata"
+	"github.com/grafeas/kritis/pkg/kritis/metadata/containeranalysis"
+	"github.com/grafeas/kritis/pkg/kritis/violation"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/sirupsen/logrus"
 )
 
 // For testing
 var (
-	podChecker    = CheckPods
 	checkInterval = 1 * time.Minute
+	podChecker    = CheckPods
+)
+
+// For testing.
+type podLister func(string) (*v1.PodList, error)
+type violationChecker func(string, v1beta1.ImageSecurityPolicy) ([]metadata.Vulnerability, error)
+type violationHandler func(string, *v1.Pod, []metadata.Vulnerability) error
+
+type Config struct {
+	PodLister         podLister
+	ViolationChecker  violationChecker
+	ViolationStrategy violation.Strategy
+}
+
+var (
+	defaultViolationStrategy = &violation.LoggingStrategy{}
 )
 
+func NewCronConfig(cs *kubernetes.Clientset, ca containeranalysis.ContainerAnalysis) *Config {
+	pl := func(ns string) (*v1.PodList, error) {
+		return cs.CoreV1().Pods(ns).List(metav1.ListOptions{})
+	}
+
+	vc := func(image string, isp v1beta1.ImageSecurityPolicy) ([]metadata.Vulnerability, error) {
+		return securitypolicy.ValidateImageSecurityPolicy(isp, "", image, ca)
+	}
+
+	cfg := Config{
+		PodLister:         pl,
+		ViolationChecker:  vc,
+		ViolationStrategy: defaultViolationStrategy,
+	}
+	return &cfg
+}
+
 // Start starts the background processing of image security policies.
-func Start(ctx context.Context) {
+func Start(ctx context.Context, cfg Config) {
 	c := time.NewTicker(checkInterval)
 	done := ctx.Done()
 
 	for {
 		select {
 		case <-c.C:
-			podChecker()
+			if err := podChecker(cfg, nil); err != nil {
```

Comment:
We can get the ImageSecurityPolicies to pass in to `podChecker` from [here](https://github.com/grafeas/kritis/blob/master/pkg/kritis/crd/securitypolicy/securitypolicy.go#L31)